sources/web/datalab/polymer/modules/Utils.ts

```diff
@@ -41,32 +37,40 @@ class Utils {
 
   /**
    * Opens a dialog with the specified options. It uses the Datalab custom element
-   * <input-dialog>, attaches a new instance to the current document, opens it
-   * and returns a promise that resolves when the dialog is closed.
-   * @param dialogOptions options for configuring the dialog
+   * according to the specified dialog type, attaches a new instance to the current
+   * document, opens it, and returns a promise that resolves when the dialog is closed.
+   * @param type specifies which type of dialog to use
+   * @param dialogOptions specifies different options for opening the dialog
    */
-  static getUserInputAsync(dialogOptions: DialogOptions) {
-    const createModal = <InputDialogElement>document.createElement('input-dialog');
-    document.body.appendChild(createModal);
+  static showDialog(type: DialogType, dialogOptions: DialogOptions) {
+    let dialogElement = '';
+    if (type === DialogType.input) {
```

Comment:
Seems like you might be able to use a "rich enum" here, basically have the dialog types be subclasses of a DialogType class, then you could just grab the dialogElement from a field of that class. But this is OK as is for just two dialog types.